### PR TITLE
ACC-786 Migrate Importer to Monolog as our current logging framework log4php is abandoned

### PR DESCRIPTION
**Pull request by georgms**

### CODE REVIEW

File: src/Command/ExportCommand.php [hunk 1]
```diff
@@ -134,11 +132,10 @@ private function configureLoggers(): void
             file_put_contents($logFile, '');
         }
 
-        $configurationAdapter = new LoggerConfigurationAdapterXML();
-        $configuration = $configurationAdapter->convert(self::LOGGER_CONFIG);
-        $configuration['appenders']['default']['params']['file'] = $logFile;
+        $logger->pushHandler(new StreamHandler($logFile));
+        $logger->pushHandler(new StreamHandler('php://stdout'));
```
Comment (georgms) [inline src/Command/ExportCommand.php:136]: :question: The log format is not exactly the same as it was defined in `Logger/import.xml` (and it is in general not possible to exactly replicate log4php messages with Monolog). Is this fine?

File: src/Command/ExportCommand.php [hunk 2]
```diff
@@ -125,7 +122,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
         return Command::SUCCESS;
     }
 
-    private function configureLoggers(): void
+    private function configureLoggers(Logger $logger): Logger
```
Comment (georgms) [inline src/Command/ExportCommand.php:125]: :wrench: Should now be called `configureLogger`.

Comment (georgms) [reply src/Command/ExportCommand.php]: Renamed.

File: src/Command/ExportCommand.php (hunk 1 above)
Comment (howard) [reply src/Command/ExportCommand.php:135]: At least I'm not aware of anyone parsing Plenty-specific log lines.

File: composer.json [hunk 3]
```diff
@@ -12,13 +12,14 @@
...
-        "vasily-kartashov/log4php": "^4.1",
         "ext-json": "*",
         "findologic/libflexport": "^2.9",
         "nesbot/carbon": "^2.38",
         "symfony/cache": "^5.1",
         "guzzlehttp/promises": "^1.4",
-        "symfony/dotenv": "^5.2"
+        "symfony/dotenv": "^5.2",
+        "monolog/monolog": "^2.3",
+        "psr/log": "^1.1"
```
Comment (howard) [inline composer.json:22]: Ideally, this library would not require any specific logger implementation, but only `psr/log`. It should be the consumer's duty to supply a compliant logger, with the freedom to choose.

**Comment by dblaichinger** [comment]
@georgms Is this PR still relevant or can it be closed? Thanks

**Comment by georgms** [comment]
* @dblaichinger I think @howard salvaged the interesting parts elsewhere. 
* @howard Can we close it?

**Comment by dblaichinger** [comment]
Closed due to inactivity, branch and PR haven been added to https://findologic.myjetbrains.com/youtrack/issue/FINDO-9216

**Pull request closed without merging**